horace_core/sqw/PixelData/@PixelTmpFileHandler/PixelTmpFileHandler.m

```diff
@@ -1,35 +1,66 @@
 classdef PixelTmpFileHandler
 
-properties (Constant, Access=private)
-    TMP_DIR_BASE_NAME_ = 'sqw_pix%05d';
-    TMP_FILE_BASE_NAME_ = '%09d.tmp';
-    FILE_DATA_FORMAT_ = 'float32';
-end
+    properties (Constant, Access=private)
+        TMP_FILE_BASE_NAME_ = 'sqw_pix%09d.tmp_sqw';
+        FILE_DATA_FORMAT_ = 'single';
+        SIZE_OF_FLOAT = 4;
+        NUM_COLS = 9;
+    end
 
-properties (Access=private)
-    tmp_dir_path_ = '';  % The path to the directory in which to write the tmp files
-    pix_id_ = -1;        % The ID of the PixelData instance linked to this object
-    has_tmp_file_ = false;  % Logical array mapping page to whether that page has a tmp file
-end
+    properties(Dependent)
+        page_size;
+        file_path;
+        pix_per_page;
+        num_pixels;
+    end
+
+    properties(GetAccess=public, SetAccess=immutable, Hidden)
+        file_path_ = '';  % The path to the directory in which to write the tmp file
+        pix_id_ = -1;     % The ID of the PixelData instance linked to this object
+        offset_ = 0;      % Offset of the filehandler's pix in file (for parallel usage)
+    end
+
+    properties
+        has_tmp_file_ = false;  % whether object has a tmp file
+    end
+
+    properties (Access=private)
+        page_size_ = get(hor_config, 'mem_chunk_size');
+        file_id_ = -1;
+    end
 
-methods
+    methods
 
-    function obj = PixelTmpFileHandler(pix_id, has_tmp_files)
+        function obj = PixelTmpFileHandler(pix_id, has_tmp_file, offset, page_size)
         % Construct a PixelTmpFileHandler object
         %
         % Input
         % -----
         % pix_id   The ID of the PixelData instance this object is linked to.
         %          This sets the tmp directory name
-        % has_tmp_files  Logical array saying if tmp exist for the given page
-        obj.pix_id_ = pix_id;
-        obj.tmp_dir_path_ = obj.generate_tmp_dir_path_(obj.pix_id_);
-        if exist('has_tmp_files', 'var')
-            obj.has_tmp_file_ = has_tmp_files;
+        % has_tmp_file  Logical array saying if tmp exist for the given page
```

Comment:
there is no indication what kind of id a pix_id should be. Filename? PixelData handle? Datum in an sqw? Needs more info

Reply:
This is a random number generated by the FileBacked entity in the range [1, 99999] (could get collisions and may want to fix this, but this is how it was done originally)
Old (Computes 5 random numbers and raises each to 10^index for some reason):
```
            obj.object_id_ = polyval(randi([0, 9], 1, 5), 10);
```
New (Computes 1 random number in range 1,99999):
```
            obj.object_id_ = randi([1, 99999], 1, 1);
```

Reply:
Probably an excellent explanation but it has not been put in the file. IF this information exists elsewhere, provide a location

Reply:
It is documented:
```
        % pix_id   The ID of the PixelData instance this object is linked to.
        %          This sets the tmp directory name
```
And on PixelDataBase
```
        object_id_;  % random unique identifier for this object, used for tmp file names
```